content/engineering/incidents/playbooks/ci.md

```diff
@@ -0,0 +1,179 @@
+# CI Playbook
+
+> This page is maintained by the [DevX Team](../../engineering/enablement/dev-experience/index.md).
+
+> Audience: any software engineer, no prior infrastructure knowlegde required.
+
+This document sums up what to do in multiple scenarios that can block the CI.
+
+See also: [Continuous Integration](https://docs.sourcegraph.com/dev/background-information/continuous_integration) in our product documentation.
+
+## Prerequisites
+
+In order to handle problems with the CI:
+
+- Have access to the `sourcegraph-ci` _project_ on Google Cloud Platform.
+  - See [#it-tech-ops](https://sourcegraph.slack.com/archives/C01CSS3TC75)
+- Have the CLI `gcloud` tool installed and have authenticated yourself.
+  - See [Gain access to the cluster](../../engineering/deployments/debugging/tutorial/index.md)
+- Set the current Kubernetes context to the `default-buildkite` _cluster_ in the `sourcegraph-ci` _project_:
+  - `gcloud config set project sourcegraph-ci`
+  - `gcloud container clusters get-credentials default-buildkite --zone us-central1-c`
+- (Optional) Install [K9s](https://k9scli.io) for easier interactions with the _pods_.
+
+## Overview
+
+The CI is what enables us to feel confident when delivering our changes to our users, and is one of the key components enabling Sourcegraph to deliver quality software. While the DevX team is in charge of managing the CI as a tool, it is essential for every engineer to be able to unblock themselves if there is a problem in order be autonomous.
+
+This page lists common failures scenarios and provide a step by step guide to get the CI back in an operational state.
+
+## Scenarios
+
+### Build has failed on the `main` branch
+
+- Gravity: _minor_
+- Impact: that commit won't be deployed on DogFood and `sourcegraph.com` until an ulterior build passes.
+- Possible causes:
+  - The `main` branch runs additional checks compared to Pull Requests builds. So it's possible that one of those checks failed.
+  - The `main` branch have changes that weren't in the Pull Request branch and those changes are causing a failure.
+  - The `main` branch is failing due to a previous build.
+
+#### Actions
+
+1. Check your build on [Buildkite](https://buildkite.com/sourcegraph/sourcegraph/builds?branch=main)
+
+- Find its link directly in the [#buildkite-main](https://sourcegraph.slack.com/archives/C02FLQDD3TQ) channel.
+- :bulb: Or run `sg ci status` in your shell, with the same branch checked out.
+
+2. Search for the failing steps, and browse the logs (:bulb: run `sg ci logs` in your shell, with the same branch checked out) .
+
+- Look for a failure explanation: it can be a test that failed or a command that return a non zero exit code.
+
+3. Check the previous builds on the `main` branch on [Buildkite](https://buildkite.com/sourcegraph/sourcegraph/builds?branch=main)
```

Comment:
We should just refer to Loki here, perhaps an example query to demonstrate searching for a message